### PR TITLE
Migrate AB Testing scripts to node

### PR DESCRIPTION
**Pull request by Jakeii**
## What does this change?
Migrate the `ab-testing` code from Deno to Node

- change any API's from deno to node
- change all the build commands from deno to pnpm
- add eslint and prettier
- switch from deno test runner to node

## Why?
As much as we like Deno, soon we'll be adding some CDK and moving the deployment of ab tests to the fastly edge dictionary to a lambda. We'd like to share the code as much as possible between the CI scripts and the lambda so migrating them to node is the best option for this.

Node now supports running typescript files without a compilation step, which makes things a bit easier!

### CODE REVIEW

File: .github/workflows/ab-testing-ui.yml [hunk 1]
```diff
@@ -41,6 +39,7 @@ jobs:
   riff-raff:
     name: AB testing Riffraff upload
     runs-on: ubuntu-latest
+    needs: build-ui
```
Comment (Jakeii) [inline .github/workflows/ab-testing-ui.yml:42]: not sure how this was working at all without this🤔
